examples/HostBuilder/Readme.MD

```diff
@@ -0,0 +1,137 @@
+```csharp
+...
+using Microsoft.Extensions.Hosting;
+
+namespace HostBuilderExample
+{
+    public class PrintHelloService : IHostedService
+    {
+        TelemetryClient _client;
+        Timer _asyncTimer;
+        public PrintHelloService(TelemetryClient telemetryClient)
+        {
+            this._client = telemetryClient;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _asyncTimer = new Timer(state =>
+            {
+                string message = "Hello Hosted Service!!!";
+                this._client.TrackEvent(new EventTelemetry(message));
+                this._client.Flush();
+            }, null, 0, 30 * 1000);
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // ... Any cleanup.
+            return Task.CompletedTask;
+        }
+    }
+}
+```
+`StartAsync` will be invoked upon the service's started; `StopAsync` will be invoked when the service is shutdown; The constructor takes TelemetryClient object as the parameter and that will be used to send data to Application Insights.
+
+In the example, as you can see, we send the message of `Hello Hosted Service!!!` to the Application Insights every 30 seconds through a timer.
+
+It could, as a matter of fact, be driven by a lot of things. For example, a message arrives in a queue; A specific event happens in EventHub; A blob shows up in a storage service and so on.
+
+## Inject the Service
+Now that we have the service, we need to inject it into the service collection. At the same time, anything required by our service, `TelemetryClient` for example, will also be registed in the inject container. Please refer the full code in [Program.cs](./src/Program.cs). You can find the outline below:
+```csharp
+...
+using Microsoft.Extensions.Hosting;
+using WebHosting = Microsoft.AspNetCore.Hosting;
+
+namespace HostBuilderExample
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            IHost host = new HostBuilder().ConfigureHostConfiguration(configHost =>
+            {
+                configHost.SetBasePath(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location));
+                configHost.AddJsonFile("hostsettings.json", optional: false);
+                configHost.AddEnvironmentVariables();
+            }).ConfigureServices((hostContext, services) =>
+            {
+                // Enable console logging for ILogger
+                services.AddLogging(cfg => { cfg.AddConsole(); });
+                // Microsoft.AspNetCore.Hosting.IHostingEnvironment is required by Application Insights SDK
+                services.AddSingleton<WebHosting.IHostingEnvironment>(provider => hostContext.HostingEnvironment.ToAspNetCoreHostingEnvironment());
+                services.AddApplicationInsightsTelemetry();
```

Comment:
I think it's a way to go and certainly helps with the scenario.
But it also brings a lot of unwanted dependencies and extra stuff to the web host.
E.g. if a user chose to have generic Host, I don't think they will be happy to get a dependency on AspNetCore.
There are things that are not applicable to generic hosts: telemetry initializers, request tracking stuff, env variables, etc...

A cleaner approach would be to either disable certain things (e.g. you can turn off requests tracking) or build DI container from scratch

Reply:
👍 I think building a DI container from the scratch sounds like a clearer approach. Let me spend some time looking into it.

Reply:
Update: Tracking the implementation by [#708](https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/708).